src/autocomplete.js

```diff
@@ -464,7 +464,7 @@ class Autocomplete {
 
     updateDocTooltip() {
         var popup = this.popup;
-        var all = popup.data;
+        var all = this.completions.filtered;
         var selected = all && (all[popup.getHoveredRow()] || all[popup.getRow()]);
```

Comment:
clearly, `popup.data` and `this.completions.filtered` don't match. but do `popup.getHoveredRow()` and `popup.getRow()` always correlate with `this.completions.filtered`?

Reply:
also it would be great to write a test which would cover this scenario to prevent a regression

Reply:
Good question. When we call [`popup.setData`](https://github.com/ajaxorg/ace/blob/master/src/autocomplete/popup.js#L237), we transparently set whatever input we give it (`this.selections.filtered` [in this case](https://github.com/ajaxorg/ace/blob/master/src/autocomplete.js#L228)) to `popup.data`. So `popup.getRow`/`popup.getHoveredRow` should correlate with `this.completions.filtered`.